src/python/pants/notes/2.5.x.md

```diff
@@ -2,6 +2,28 @@
 
 See https://www.pantsbuild.org/v2.5/docs/release-notes-2-5 for an overview of the changes in this release series.
 
+## 2.5.0.dev2 (Apr 16, 2021)
+
+### New Features
+
+* Handle URL requirements with constraints files. ([#11907](https://github.com/pantsbuild/pants/pull/11907))
+
+* Release Pants with Python 3.9 ([#11858](https://github.com/pantsbuild/pants/pull/11858))
+
+* Allow Python files without a file ending ([#11905](https://github.com/pantsbuild/pants/pull/11905))
```

Comment:
Bug fix

Reply:
Also disagree. I mean any missing feature can be recast as a bug, so it's a bit of a grey area, but to me bug means "not working as intended", and this was clearly working as intended, it's just that the intention was wrong...

We explicitly coded that extension check, we just failed to think about the cases where it would not meet user needs. That's not the same as a "bug" in my book.

Reply:
I view it as a bug in terms of our design, ack that it is not a bug in the actual implementation. When I wrote the target definition, I fully meant to support Python scripts, but due to lack of foresight, did not handle an edge case.

Reply:
NBD either way though